src/hotspot/cpu/riscv/templateTable_riscv_64.cpp

```diff
@@ -240,48 +240,59 @@ void TemplateTable::sipush() {
 void TemplateTable::ldc(bool wide) {
   Register Rscratch1 = R5_scratch1,
            Rscratch2 = R6_scratch2,
-           Rcpool    = R3_ARG1_PPC;
+           Rcpool    = R7_scratch3;
 
   transition(vtos, vtos);
-  Label notInt, notFloat, notClass, exit;
+  Label notInt, notFloat, notClass, isClass, exit;
 
   __ get_cpool_and_tags(Rcpool, Rscratch2); // Set Rscratch2 = &tags.
   if (wide) { // Read index.
     __ get_2_byte_integer_at_bcp(1, Rscratch1, InterpreterMacroAssembler::Unsigned);
   } else {
-    __ lbz_PPC(Rscratch1, 1, R22_bcp);
+    __ lb(Rscratch1, R22_bcp, 1);
   }
 
   const int base_offset = ConstantPool::header_size() * wordSize;
   const int tags_offset = Array<u1>::base_offset_in_bytes();
 
   // Get type from tags.
-  __ addi_PPC(Rscratch2, Rscratch2, tags_offset);
-  __ lbzx_PPC(Rscratch2, Rscratch2, Rscratch1);
+  __ add(Rscratch2, Rscratch1, Rscratch2);
+  __ lb(Rscratch2, Rscratch2, tags_offset);
 
-  __ cmpwi_PPC(CCR0, Rscratch2, JVM_CONSTANT_UnresolvedClass); // Unresolved class?
-  __ cmpwi_PPC(CCR1, Rscratch2, JVM_CONSTANT_UnresolvedClassInError); // Unresolved class in error state?
-  __ cror_PPC(CCR0, Assembler::equal, CCR1, Assembler::equal);
+  __ addi(Rscratch2, Rscratch2, -JVM_CONSTANT_UnresolvedClass);
+  __ beqz(Rscratch2, isClass);
+  __ addi(Rscratch2, Rscratch2, JVM_CONSTANT_UnresolvedClass);
```

Comment:
Can we avoid adding the constant back after taking a difference? 
Since only compare result is interesting, is possible to allocate a new temp register for difference result?